### DISCUSSION TITLE
Acessing Sails API endpoint from another react website

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [neonexus/sails-react-bootstrap-webpack#133] [General]
@neonexus,

I would like your take on the below use case:

I have this Sails.js-React webapp with MySQL DB. For a particular scenario, I have to launch another website (also React front-end but it does not have a backend) from this web app & also need to pass a JSON file to it while launching.

JSON files are stored in MySQL to which Sails has access to. The size of a JSON file would be max 10 MB.

Could you please suggest an effective approach to deal with this? Thanks.

**Comment by neonexus** [comment]
This is an odd scenario... I don't know if I can provide effective guidance on this. 

The user is going to have to wait until the second site has loaded the JSON from the first. The second site is going to have to make a request to the first to load the JSON on the client.

**Reply by MaheshkumarSundaram** [reply]
@neonexus,

If the other site is a trusted one, can't I just expose the API endpoint (GET) to serve the JSON from MySQL through `allowOrigins` to include that specific site in `config/security.js`?

Thank you.

**Reply by neonexus** [reply]
@MaheshkumarSundaram sorry, this fell through the cracks. I would recommend the `allowOrigns` mixed with a shared secret for a bit more protection.

**Reply by neonexus** [reply]
Did this ever get resolved?

**Reply by MaheshkumarSundaram** [reply]
@neonexus Yes. I have used `allowOrigins` for that specific route alone. I'll add a seret token too as you suggested. Thank you.
